File: src/main/java/ch/wisv/choice/exam/service/ExamService.java
package ch.wisv.choice.exam.service;


import ch.wisv.choice.course.model.Course;
import ch.wisv.choice.exam.model.Exam;

import java.util.Collection;

public interface ExamService {

    void createExam(Exam exam);

    Collection<Exam> readExams();

    Collection<Exam> readExamsByCourse(Course course);

    void updateExam(Exam exam);

    void deleteExam(Exam exam);

    void deleteExamsByCourse(Course course);
}

File: src/main/kotlin/ch/wisv/choice/exam/service/ExamService.kt
package ch.wisv.choice.exam.service


import ch.wisv.choice.exam.model.Exam

interface ExamService {

    fun createExam(exam: Exam): Exam

    fun getExams(): Collection<Exam>

    fun getExamsByCourse(code: String): Collection<Exam>

    fun getExamById(id: Long): Exam

    fun deleteExam(exam: Exam)

    fun deleteExamsByCourse(code: String)
}
